### PR TITLE
Rpm defaults fix for 4.3

### PR DESCRIPTION
**Pull request by Asizon**
-Removed non used rpm defaults from 4.2 to avoid msp complications with outdated and not suported version by this.
-Allows confirm or decline dyn values changes.

### CODE REVIEW

**Comment by Asizon** [comment]
This feature is added here for 10.8.0, removed it for 4.2 version since it is not in 10.7.0:
https://github.com/betaflight/betaflight-configurator/pull/1955

**Comment by ctzsnooze** [comment]
Thanks, Asizon.

With the new dynamic notch defaults, many quads can fly safely without RPM filtering, and without making any changes to dynamic notch settings.

I wonder now if we might just provide a warning note, and not make any changes to the dynamic notch values.

**Review by haslinghuis — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by Asizon** [comment]
@ctzsnooze @KarateBrot I will make flight test and logs this week to see graphicaly if we really need this rpm defaults now with the new dynamic notch.

**Comment by ctzsnooze** [comment]
I had feedback from Karatebrot that we should continue to make these changes.

**Comment by ctzsnooze** [comment]
I tested it, and the code works really well.

If the dynamic notch filters are at default, the user gets 'safe' notch filters when turning RPM filtering off, and 'propwash friendly' notch filters when turning RPM on.  For both transitions, they get this message:

```
WARNING: Some dynamic notch values have been changed to default values because the RPM filtering has been activated/deactivated.
Please, check before flying.
```

However, if the user has either loaded a preset that changed their dynamic notch filter settings, or themselves made a change, they get no warning message when turning RPM filtering off, and will not get the 'safer' filters.

Maybe this is a bit illogical?  It seems strange to show a strong warning when we shift to 'safe' filters, but we show no warning, none at all, when the user turns RPM off, and we do not provide the 'safer' filters. 

This is a tough one.  

The 'unsafe' transition is from RPM to non-RPM.  That's when a user needs strong dynamic notch filtering, or they may cook their motors.  

It seems to me that we should always show a warning in this case.  However the warnings should be different, depending on whether or not we automatically changed the dynamic notches.

For example, when we don't automatically change the dynamic notch filter settings:
```
WARNING: Turning off RPM filtering may overheat the motors.  Because you have custom gyro dynamic notch settings, we do not automatically change those values.  You should check that they are appropriate for a non-RPM filter configuration.  Typically, three or more dynamic notches are required. Please monitor your motor temperatures closely for the next few flights.
```

Whereas if we do automatically change the dynamic notch filter settings:
```
WARNING: Turning off RPM filtering may overheat the motors.  Typically three, wider dynamic notches are required.  We have automatically made those changes for you.  Please monitor your motor temperatures closely for the next few flights.
```

If we enable RPM filtering, the downside of not changing dynamic notches is only that you may have stronger filtering than is required, leading to a bit of propwash.  I'm not sure we really need a warning.  A note should be sufficient.

For example, on enabling RPM filtering, where we do not automatically change dynamic notch settings:
```
NOTE: Turning on RPM filtering usually means that a single, narrower dynamic notch filter is sufficient.  Because you have custom gyro dynamic notch settings, we have not automatically made that change.  Please review your dynamic notch settings before you fly, and monitor your motor temperatures for the next few flights.
```

And when we do:
```
NOTE: Turning on RPM filtering usually means that a single, narrower dynamic notch filter is sufficient.  We have automatically made that change for you.  Monitor your motor temperatures carefully for the next few flights.
```

**Comment by Asizon** [comment]
@ctzsnooze thanks for your coments! I can easy add a diferent dialog for this nothing change case yes.

**Comment by Asizon** [comment]
Here my flight Logs with 3 cases, hope that they can be useful @KarateBrot @ctzsnooze 

[NO RPM_DefaultDynfilter.zip](https://github.com/betaflight/betaflight-configurator/files/7830283/NO.RPM_DefaultDynfilter.zip)
[RPM_DefaultDynFilter.zip](https://github.com/betaflight/betaflight-configurator/files/7830284/RPM_DefaultDynFilter.zip)
[RPM_DynCount-1_Q-500.zip](https://github.com/betaflight/betaflight-configurator/files/7830285/RPM_DynCount-1_Q-500.zip)

**Review by blckmn — APPROVED** [review]

**Comment by ctzsnooze** [comment]
In the past, when turning RPM filtering off, we:
- got a warning message, 
- automatically shifted to safer notch filter settings

With this PR, when we turn RPM tilting off, we DO NOT get safer filters, and we DO NOT get a warning.
I think this is unsafe.

Also, in the past, when enabling RPM filtering, we
- got a warning
- automatically shifted to notch filter to a  configuration that caused less delay.

With this PR, we do not get a warning, and we do not get the filter change.

These automatic behaviours have been existing for a long time, and have worked well.  I am very concerned about changing the behaviour.

As I said before, if we are not going to make the automatic change, we need to show a message that warns the user of the bad things that could happen, and what they need to do.

**Comment by ctzsnooze** [comment]
@Asizon thanks for your logs.
This comparison:
- Left side, RPM filtering + 3 Notches, Q=300 (what you would get with this PR on enabling RPM with no change in DN)
- Right side, RPM Filtering + 1 Notch, Q=500 (our current auto configuration when enabling RPM)
As expected there is a 35hz resonant peak in the spectrum due to significant propwash problems from filter delay.
Screen Shot 2022-01-10 at 18 53 45

**Comment by ctzsnooze** [comment]
This comparison is:
- left side: RPM filter ON, with a single notch Q=500 (our auto setting for RPM enabled)
- right side: RPM filter OFF, three notches Q=300 (our auto setting for RPM disabled)
Note more high frequency noise on the right, though less propwash.
It's not hard to imagine how much worse the noise would be if we only had one single dynamic notch at say 450Hz, and no RPM filtering, which could happen *without warning*  if the user just turned RPM filtering off in this PR.
Screen Shot 2022-01-10 at 18 58 36

**Comment by ctzsnooze** [comment]
Our automatic changes have, in the past, given users reasonable filter delay when enabling RPM filtering, and safe filters when turning it off.  If we remove these automatic changes, we run a risk that people will end up with too much propwash, or unsafe filters, and not really understand how that happened.  All I have to do is modify my dynamic notch setting a little bit, and suddenly I lose that protection.  To me this isn't safe.

It would be OK if we had a strong warning message as above.  It would be even better if that message included an 'apply changes' button, so the user could agree to the automatic change.

**Comment by Asizon** [comment]
@ctzsnooze these automatic changes are existing from 4.3,not long time, but this pr still continues changing dyn defaults when rpm enabled/disabled. Only doesnt change if dyn values are already changed manually or with presets.

**Comment by Asizon** [comment]
Thankyou @ctzsnooze ,i need to take time tomorrow and this weekend to make these changes amd some others.

**Comment by Asizon** [comment]
I have made requested changes @ctzsnooze . Thanks @limonspb for the new yes/no dialog and pointed it.
Used generalized i18n texts to avoid more increasses of maintenance.

File: locales/en/messages.json
Comment (ctzsnooze) [inline locales/en/messages.json:1515]: One single message is fine.
Maybe add the word `/effectiveness`, so it reads: `increasing/decreasing filter delay/effectiveness.`

File: src/js/tabs/pid_tuning.js
Comment (ctzsnooze) [inline src/js/tabs/pid_tuning.js:454]: Is there a way for Agree / Cancel buttons to be on the right side of the dialog box?

**Comment by ctzsnooze** [comment]
Works perfectly apart from one issue that I noticed.  The dialog only appears the first time the toggle is moved.  So you could, for instance, have this result:

User changes DShot Telemetry from ON to OFF -> dialog opens (good).
User Agrees -> Filters change (good)
Then, without saving, user move DShot Telemetry from OFF to ON -> dialog DOES NOT open (bad).
Result = dynamic idle settings for OFF situation, but RPM filter is on.

Otherwise this is really good - simple and intuitive.

File: locales/en/messages.json
Comment (Asizon) [reply locales/en/messages.json:1515]: I will change it :)

File: src/js/tabs/pid_tuning.js
Comment (Asizon) [reply src/js/tabs/pid_tuning.js:454]: Yes, but it is using standarized dialog added in presets tab, you can try to pick a preset, is the same dialog.

File: locales/en/messages.json
Comment (haslinghuis) [inline locales/en/messages.json:1515]:
```suggestion
        "message": "<span class=\"message-negative\"><b>WARNING: This change will enable/disable RPM filtering, increasing/decreasing filter delay delay/effectiveness.</b></span><br><br>Reset the dynamic notch filters to recommended values?"
```

Please add second <br>

File: src/js/tabs/pid_tuning.js
Comment (limonspb) [reply src/js/tabs/pid_tuning.js:454]: yeah, that probably should be a separate PR if we want to change how this dialog looks.

**Comment by Asizon** [comment]
@ctzsnooze Understanding your suspicions, this happens only in the filters tab and I don't know what can be the best way to approach it. The motors tab works fine since the save button is disabled preventing changes from being saved

**Comment by ctzsnooze** [comment]
We do need a fix in the filters tab, because otherwise, a user with RPM on, and who turns it off in filters tab, will get 'stuck' with the wrong (rpm 'off') values even if they switch it back on - and most people won't realised that the values are wrong.

**Comment by Asizon** [comment]
Yes @ctzsnooze i will check how we can fix that behabiour.

**Comment by ctzsnooze** [comment]
@Asizon it would be great to have this fixed soon, if at all possible.

**Comment by Asizon** [comment]
@ctzsnooze i will store dyn notch values and push then if no dialog showsup

**Comment by Asizon** [comment]
Done @ctzsnooze this is ready and finished now.

**Review by ctzsnooze — DISMISSED** [review]

**Comment by ctzsnooze** [comment]
Works really well now - thank you!  Approved.

**Comment by Asizon** [comment]
@ctzsnooze sonarcloud issues are arround the file. They can be fixed in future PRs, not really important

**Review by KarateBrot — APPROVED** [review]

**Review by KarateBrot — APPROVED** [review]

**Review by haslinghuis — APPROVED** [review]

**Review by ctzsnooze — APPROVED** [review]

**Review by limonspb — APPROVED** [review]

**Pull request merged**
